docs/entityoperator.md

```diff
@@ -0,0 +1,234 @@
+- Kafka controllers may still be running (if a quorum is maintained).
+
+Replication between brokers (spread across clusters) continues as long as there’s a majority quorum of controllers.
+
+- ✅ If a majority of controllers are still available in the surviving clusters, Kafka continues running.
+- 🚨 If the majority of controllers were in the central cluster and lost, Kafka will experience a complete outage.
+
+💡 Mitigation -> user need to ensure the `controller.quorum.voters` are spread across clusters to avoid losing the majority.
+
+## Can Kafka Clients Still Communicate with Brokers?
+
+Yes, if at least one broker remains reachable.
+
+Clients connect to brokers, not the Entity Operator or the central cluster.
+
+- If clients' bootstrap servers list includes brokers running in the surviving clusters, they can still connect.
+- If a leader partition for a topic is hosted on a broker in a surviving cluster, the topic remains available.
+- If all leader partitions for a topic were on brokers in the central cluster, that topic becomes unavailable.
+
+💡 Mitigation -> Use Kafka’s rack awareness (`broker.rack`) and ISR (In-Sync Replica) balancing to distribute leader partitions across clusters.
+
+## Can Clients Still Produce & Consume Messages?
+
+Yes, if leader partitions are available.
+No, if all leader partitions were in the lost central cluster.
+
+**Producing Messages**
+
+- A producer can send messages only if the leader partition for a topic is still available in a surviving cluster.
+- If the leader was in the lost central cluster, Kafka automatically elects a new leader (if ISR exists).
+- If no ISR exists, production fails.
+
+**Consuming Messages**
+
+- Consumers can still fetch messages from available partitions.
+- If consumer group metadata was stored in a lost broker, the group might experience issues.
+
+💡 Mitigation -> Enable `min.insync.replicas` and leader election across clusters to ensure partition availability.
+
+
+## What About Entity Operator Functions?
+The Entity Operator becomes unavailable when the central cluster goes down. However, this does not impact existing Kafka clients directly because
+
+- Kafka clients do not interact with the Entity Operator at runtime.
+- User authentication still works as long as secrets (TLS/SCRAM) were distributed to all clusters.
+- Topics and ACLs remain intact but cannot be updated or created until the central cluster recovers.
+
+Kafka clients (producers and consumers) can still authenticate and connect to Kafka brokers in the surviving clusters as long as the necessary authentication credentials (secrets) are available in those clusters.
+
+If the KafkaUser secrets only exist in the central cluster, then when it goes down, clients in other clusters cannot authenticate to Kafka brokers. However, if these secrets were already copied to all clusters where brokers are running, authentication will still work even if the central cluster is down.
```

Comment:
Is the issue that clients will loose access to the secrets?? or are you implying that Kafka can not authenticate user requests. as brokers use secrets to performa auth?